packages/synapse-react-client/src/components/HasAccess/HasAccessV2.tsx

```diff
@@ -137,10 +187,18 @@ export function HasAccessV2(props: HasAccessProps) {
     AccessRequirement[]
   >([])
 
-  const { entityId, showButtonText = true } = props
-
+  const {
+    entityId,
+    showButtonText = true,
+    hideAccessIconForExternalFileHandle,
+  } = props
   const restrictionUiTypeValue = useGetRestrictionUiType(entityId)
 
+  const { isExternalFileHandle, isLoading } = useGetIsExternalFileHandle(
+    entityId,
+    { enabled: hideAccessIconForExternalFileHandle ?? false },
+  )
```

Comment:
Simplify & rename `isLoading` for clarity
```suggestion
  const { isExternalFileHandle, isLoading: isLoadingIsExternalFileHandle } = useGetIsExternalFileHandle(
    entityId,
    { enabled: hideAccessIconForExternalFileHandle },
  )
```